### PR TITLE
Ivy bce with logits

### LINKED ISSUE #7295: ivy.binary_cross_entropy_with_logits
This issue is raised as a result of the discussion [here](https://github.com/unifyai/ivy/pull/7076#issuecomment-1324023470).

**Why should this be implemented?**
Because 2 of the native frameworks have it, and it's a complex function.

**Links to native framework implementations**
- [PyTorch](https://pytorch.org/docs/stable/generated/torch.nn.functional.binary_cross_entropy_with_logits.html#torch.nn.functional.binary_cross_entropy_with_logits)
- [TensorFlow](https://www.tensorflow.org/api_docs/python/tf/nn/weighted_cross_entropy_with_logits)(with positional weights)
- [Tensorflow](https://www.tensorflow.org/api_docs/python/tf/nn/sigmoid_cross_entropy_with_logits)(without positional weights)

### PR DESCRIPTION
**Pull request by bipinKrishnan**
Adds binary cross entropy with logits loss to Ivy functional API.

Closes #7295

### CODE REVIEW

**Comment by bipinKrishnan** [comment]
@johanjino The test function written right now covers the case were some values are passed to ```pos_weight```, but it doesn't cover the scenario of ```pos_weight=None```. I'm not sure if I should write a separate test to cover this case or modify the current test function somehow. Any pointers on this would be great :slightly_smiling_face:

**Comment by johanjino** [comment]
Hey @bipinKrishnan , you could create a new helper function within the module and use it to generate test data which includes `None`. This might be a steep learning curve and you can refer to https://lets-unify.ai/ivy/deep_dive/ivy_tests.html. 

If it seems too difficult please let me know, this work is mergeable and the `None` testing can be added later on too!

**Comment by bipinKrishnan** [comment]
Thanks for the quick feedback @johanjino.

I've added the necessary helper function to test with ```None``` values. Let me know if anything is missing from the implementation.

**Comment by bipinKrishnan** [comment]
@johanjino Not sure what happened, but some tests are failing on github which are passing on my local machine:
Screenshot from 2022-11-29 11-26-56

**Comment by johanjino** [comment]
Hey @bipinKrishnan , the array_with_none seems well defined. The Error is due to a falsifying example. This is probably not seen in your machine because of how Hypothesis draws examples. You can view it in the checks.

**Comment by johanjino** [comment]
Do let me know, if you find it difficult to fix it or if you face any other issue

**Comment by bipinKrishnan** [comment]
@johanjino While going through the outputs of the failing tests, I found that the tests for bce with logits loss and other loss functions already implemented are failing due to the same error. Does this mean that the error is due to some other code present in Ivy? I'm not completely sure about this, but any pointers on how to move forward would be awesome :)

**Comment by bipinKrishnan** [comment]
@johanjino The tests for bce with logits loss are passing. Can we merge this PR now?

**Comment by johanjino** [comment]
Hey @bipinKrishnan , sorry for the delay, was a litte caught up last week. This seems great I will merge the PR now! Thank you!
